src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,106 @@
-import { useState } from 'react';
+/* eslint-disable max-len */
+import React, { useEffect, useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+interface Props {
+  onAdd: (movie: Movie) => void;
+}
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
```

Comment:
```suggestion
  const [newMovie, setNewMovie] = useState({
     title: '',
     description: '',
     ...
  });
```
make one state for all form fields. use event.target.name for change proper field. pay attention that input should have attribute name should be the same as appropriate newMovie field.